### PR TITLE
[MD] Add experimental callout for index pattern section

### PR DESCRIPTION
**Pull request by yibow98**
Signed-off-by: Yibo Wang <yibow@amazon.com>

### Description
This PR fixes several issues from https://github.com/opensearch-project/ux/issues/16

Index pattern listing page:
Screen Shot 2022-10-05 at 8 07 54 PM

Index pattern create page-step one:
Screen Shot 2022-10-05 at 8 05 54 PM
Index pattern create page-step two:
Screen Shot 2022-10-05 at 8 06 13 PM
Index pattern create page-step three:
Screen Shot 2022-10-05 at 8 06 38 PM

 
### Issues Resolved
https://github.com/opensearch-project/ux/issues/37
https://github.com/opensearch-project/ux/issues/39
https://github.com/opensearch-project/ux/issues/41
https://github.com/opensearch-project/ux/issues/43

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx
Comment (kristenTian) [inline src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx:247]: I though kroosh@ mentioned no listing page?

File: src/plugins/index_pattern_management/public/components/create_index_pattern_wizard/create_index_pattern_wizard.tsx
Comment (kristenTian) [inline src/plugins/index_pattern_management/public/components/create_index_pattern_wizard/create_index_pattern_wizard.tsx:356]: could you add this behind this switch `dataSourceEnabled`. Only display the banner when true

File: src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx
Comment (yibow98) [reply src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx:247]: Just index pattern detail page will be excluded. Listing page and create page will both have callout

**Comment by seraphjiang** [comment]
could we add screenshot?

**Comment by kavilla** [comment]
Thanks @yibow98, looks like the snapshots need to be updated for tests since they are failing. Which can be done with `yarn test:jest -u`.

File: src/plugins/index_pattern_management/public/components/create_index_pattern_wizard/create_index_pattern_wizard.tsx
Comment (noCharger) [reply src/plugins/index_pattern_management/public/components/create_index_pattern_wizard/create_index_pattern_wizard.tsx:356]: @yibow98  https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/index_pattern_management/public/plugin.ts#L51 Start from here~

File: src/plugins/index_pattern_management/public/components/experimental_callout/experimental_callout.tsx
Comment (yibow98) [reply src/plugins/index_pattern_management/public/components/experimental_callout/experimental_callout.tsx]: Done

**Comment by yibow98** [comment]
> Thanks @yibow98, looks like the snapshots need to be updated for tests since they are failing. Which can be done with `yarn test:jest -u`.

Thanks @kavilla ! I updated snapshots accordingly.

File: src/plugins/index_pattern_management/public/components/create_index_pattern_wizard/create_index_pattern_wizard.tsx
Comment (yibow98) [reply src/plugins/index_pattern_management/public/components/create_index_pattern_wizard/create_index_pattern_wizard.tsx]: @noCharger Thanks! Done.

**Review by kristenTian — DISMISSED** [review]

**Review by noCharger — APPROVED** [review]
LGTM

**Review by kristenTian — APPROVED** [review]

**Review by zhongnansu — APPROVED** [review]

File: CHANGELOG.md
Comment (ashwin-pc) [inline CHANGELOG.md:19]: Just one changelog entry should suffice

File: src/plugins/index_pattern_management/public/components/experimental_callout/experimental_callout.tsx
Comment (ashwin-pc) [inline src/plugins/index_pattern_management/public/components/experimental_callout/experimental_callout.tsx:34]: I havent had the chance to use the localization features yet but calling them like this would only run localization once. If we switch our locale after the initial page load, these string will not change.

Why dont we use the `FormattedMessage` component from `@osd/i18n/react` instead of declaring each string separately like this?

File: src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx
Comment (ashwin-pc) [inline src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx:247]: Looks like we may not need this logic since the list should not be rendered in the first place if `dataSourceEnabled` is false. See line 221

**Review by ashwin-pc — COMMENTED** [review]
Nice change, just a few comments

File: src/plugins/index_pattern_management/public/components/experimental_callout/experimental_callout.tsx
Comment (mpabba3003) [reply src/plugins/index_pattern_management/public/components/experimental_callout/experimental_callout.tsx:34]: This will be changed in next PR

File: CHANGELOG.md
Comment (yibow98) [reply CHANGELOG.md:19]: Fixed.

File: src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx
Comment (yibow98) [reply src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx:247]: If you look at line 220, those 2 empty state components are returned only when there are NO index patterns in the table. If there are any index patterns created earlier or using default, then the table component would be displayed.
So my condition is that, if table shows up & data source is enabled, it should display experimental call out.
When index patterns length is 0, then create index pattern prompt will show up and when user clicks on create button, on create page experimental call out will be displayed.

**Review by kristenTian — APPROVED** [review]

File: src/plugins/index_pattern_management/public/components/experimental_callout/experimental_callout.tsx
Comment (mpabba3003) [reply src/plugins/index_pattern_management/public/components/experimental_callout/experimental_callout.tsx:34]: #2543

**Review by KrooshalUX — APPROVED** [review]
Purely from UX perspective based on screenshots, lgtm!

File: src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx
Comment (ashwin-pc) [reply src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx:247]: Oh okay, if thats the case, a unit test here would be more useful than the snapshot test you have for the `ExperimentalCallout` component. You want to be testing logic and not the text itself since that may change and that will be intentional.

Will not block on it though.

Comment (yibow98) [reply src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx:247]: Agree. We will have another PR for unit tests coverage since we were working on several same files. And once these two PRs has been merged, I will start to update unit test accordingly.

**Review by ashwin-pc — APPROVED** [review]

**Review by ashwin-pc — APPROVED** [review]

**Review by kristenTian — APPROVED** [review]

**Pull request merged**
